json/src/de.rs

```diff
@@ -205,12 +207,30 @@ impl<R: Read> DeserializerImpl<R> {
                 visitor.visit_str(s)
             }
             b'[' => {
+                self.remaining_depth -= 1;
```

Comment:
I suppose all this could be moved into the `SeqVisitor` but I'm not sure if that'd make things any more clean.